### PR TITLE
Initial Client

### PR DESCRIPTION
**Pull request by andrewls**
This completes the initial API side of the conversica integration. We should probably expect to continue making changes here as necessary.

### CODE REVIEW

File: .byebug_history [hunk 1]
```diff
@@ -0,0 +1,5 @@
+quit
```
Comment (andykallhoff) [inline .byebug_history]: I don't understand this file

File: bin/console [hunk 2]
```diff
@@ -7,8 +7,8 @@ require "conversica"
...
 
 # (If you use this, don't forget to add pry to your Gemfile!)
-# require "pry"
-# Pry.start
+require "pry"
+Pry.start
 
-require "irb"
-IRB.start(__FILE__)
+# require "irb"
```
Comment (andykallhoff) [inline bin/console]: remove these?

File: lib/conversica/client/lead.rb [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      def validate_type
+        PERMITTED_COLUMNS.each do |k|
+          permitted_classes, error = if BOOLEAN_COLUMNS.include?(k)
+                                       [
+                                         [NilClass, TrueClass, FalseClass],
+                                         "#{k} must be a boolean value or null"
+                                       ]
+                                     elsif DATE_COLUMNS.include?(k)
+                                       [
+                                         [NilClass, Date, Time],
```
Comment (andykallhoff) [inline lib/conversica/client/lead.rb:55]: do we want to add `DateTime` ?

File: lib/conversica/client/lead.rb [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      end
+
+      def validate_required
+        REQUIRED_ATTRIBUTES.each do |k|
+          self.errors << "#{k} is required" if self.attributes[k].blank?
+        end
+      end
+
+      # I am perhaps prouder of this method name than I should be
+      def conversicate
```
Comment (andykallhoff) [inline lib/conversica/client/lead.rb:82]: 👏 agreed

**Review by andykallhoff — APPROVED** [review]
Nice looks great, mainly just had checks/questions. Great stuff as usual

File: .byebug_history (hunk 1 above)
Comment (andrewls) [reply .byebug_history]: It should probably be in the gitignore. It's just a list of the most recent commands I ran with byebug. Byebug uses it in its feature where you can push the up arrow and it cycles through the previous commands you ran. I'll remove it.

File: lib/conversica/client/lead.rb (hunk 3 above)
Comment (andrewls) [reply lib/conversica/client/lead.rb:55]: `DateTime` is a subclass of `Date`, so if you check if `DateTime.now.is_a?(Date)` it returns true, and `Date === DateTime.now` is also true

**Pull request merged**